modules/apps/29-fee/keeper/keeper.go

```diff
@@ -178,6 +174,49 @@ func (k Keeper) GetAllRelayerAddresses(ctx sdk.Context) []*types.RegisteredRelay
 	return registeredAddrArr
 }
 
+// SetForwardRelayerAddress sets the forward relayer address during OnRecvPacket in case of async acknowledgement
+func (k Keeper) SetForwardRelayerAddress(ctx sdk.Context, packetId *channeltypes.PacketId, address string) {
+	store := ctx.KVStore(k.storeKey)
+	store.Set(types.KeyForwardRelayerAddress(packetId), []byte(address))
+}
+
+// GetForwardRelayerAddress gets forward relayer address for a particular packet
+func (k Keeper) GetForwardRelayerAddress(ctx sdk.Context, packetId *channeltypes.PacketId) (string, bool) {
+	store := ctx.KVStore(k.storeKey)
+	key := types.KeyForwardRelayerAddress(packetId)
+
+	if !store.Has(key) {
+		return "", false
+	}
+
+	addr := string(store.Get(key))
+	return addr, true
+}
+
+// GetAllForwardRelayerAddresses returns all forward relayer addresses stored for async acknowledgements
+func (k Keeper) GetAllForwardRelayerAddresses(ctx sdk.Context) []*types.ForwardRelayerAddress {
+	store := ctx.KVStore(k.storeKey)
+	iterator := sdk.KVStorePrefixIterator(store, []byte(types.ForwardRelayerPrefix))
+	defer iterator.Close()
+
+	var forwardRelayerAddr []*types.ForwardRelayerAddress
+	for ; iterator.Valid(); iterator.Next() {
+		keySplit := strings.Split(string(iterator.Key()), "/")
+
+		seq, _ := strconv.ParseUint(keySplit[3], 0, 64)
```

Comment:
Wasn't sure if/how we should handle this error. If this errs the code is broken elsewhere.

Reply:
I think you could probably just panic. I'd say it's nicer than just ignoring the error, which could maybe be considered "bad code hygiene", and it seems to be done that way in other parts of the core codebase.

Reply:
Just panicing makes sense here. This isn't run within the state machine anyway. So it actually is preferable to just panic if someone runs this command and finds this issue out

Reply:
Added a panic. Not sure about the error message.